daemon/response.go

```diff
@@ -67,11 +68,26 @@ type resp struct {
 	Maintenance *errorResult `json:"maintenance,omitempty"`
 }
 
-func (r *resp) transmitMaintenance(kind client.ErrorKind, message string) {
-	r.Maintenance = &errorResult{
-		Kind:    kind,
-		Message: message,
-	}
+func maintenanceForRestartType(rst state.RestartType) *errorResult {
+	e := &errorResult{}
+	switch rst {
+	case state.RestartSystem, state.RestartSystemNow:
+		e.Kind = client.ErrorKindSystemRestart
+		e.Message = daemonRestartMsg
+	case state.RestartDaemon:
+		e.Kind = client.ErrorKindDaemonRestart
+		e.Message = systemRestartMsg
+	case state.RestartSocket:
+		e.Kind = client.ErrorKindDaemonRestart
+		e.Message = socketRestartMsg
+	case state.RestartUnset:
+		// don't set anything as unset means it's not really an errorResult here
```

Comment:
we don't expect this case anymore, do we? we should panic instead?

Reply:
well ServeHTTP still calls it for setting Maintenance inside responses, so I'm not sure, I think returning nil here still makes sense, but I can do this in ServeHTTP instead:

```go
	if rsp, ok := rsp.(*resp); ok {
		_, rst := st.Restarting()
		if rst != state.RestartUnset {
			rsp.Maintenance = maintenanceForRestartType(rst)
		}

```

Reply:
well we can revert if it makes more sense to return nil, but I will push up a commit making this panic and adjusting ServeHTTP appropriately.